File: C++/2-Medium/00137._只出现一次的数字_II/00137._只出现一次的数字_II.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int bitCounts[32] = {0}; // 用于存储32位整数的每一位的总和
        for (int num : nums) {
            for (int i = 0; i < 32; ++i) {
                bitCounts[i] += (num >> i) & 1; // 更新每一位的和
            }
        }
        int result = 0;
        for (int i = 0; i < 32; ++i) {
            result |= (bitCounts[i] % 3) << i; // 对每一位进行处理，并加到结果中
        }
        return result;
    }
};

// 主函数用于测试
int main() {
    Solution solution;
    vector<int> nums = {2,2,3,2}; // 示例
    cout << "The single number is: " << solution.singleNumber(nums) << endl;
    return 0;
}

File: Python/2-Medium/00137._只出现一次的数字_II/00137._只出现一次的数字_II.py
from typing import List

def singleNumber(nums: List[int]) -> int:
    """
    找出数组中唯一一个不是出现三次的数字。
    
    参数:
    nums: 包含整数的列表，其中除了一个元素出现一次外，其他元素都出现三次。
    
    返回:
    int: 只出现一次的那个数字。
    
    示例:
    >>> singleNumber([2,2,3,2])
    3
    >>> singleNumber([0,1,0,1,0,1,99])
    99
    """
    once, twice = 0, 0
    for num in nums:
        # 更新once和twice
        once = ~twice & (once ^ num)
        twice = ~once & (twice ^ num)
    return once

# 测试示例
test_cases = [
    ([2, 2, 3, 2], 3),
    ([0, 1, 0, 1, 0, 1, 99], 99),
]

# 运行测试用例
results = [(nums, singleNumber(nums)) for nums, expected in test_cases]

results